starboard/build/config/BUILD.gn

```diff
@@ -145,10 +145,6 @@ config("starboard") {
       defines += [ "INTERNAL_BUILD" ]
     }
 
-    if (sb_allows_memory_tracking) {
```

Comment:
Thanks for catching, yes ! Added an extra commit that removes it.

Note, i'm intentionally _not_ removing the relevant Starboard functions, in `starboard/memory_reporter.h`, because i missed the window for SB 15 changes. So these will just sit entirely unused for an extra year.